pacman/model/graphs/abstract_multiple_partition.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) 2019-2020 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from collections import OrderedDict
+
+from pacman.exceptions import PacmanConfigurationException
+from spinn_utilities.ordered_default_dict import DefaultOrderedDict
+from spinn_utilities.ordered_set import OrderedSet
+
+
+class AbstractMultiplePartition(object):
+
+    def __init__(self, pre_vertices):
+        self._pre_vertices = OrderedDict()
```

Comment:
I would use either pre and post OR source and destination. The mix is confusing.

Reply:
its prevertices as the only thing that handles multiple pre is the src segged which this api allows to work in a general case. 

pre vertices is what the graph handles, so keeping in the same vain makes it sensible there. multiple dests are handled by the edges of a partition.